src/ontodia/diagram/view.ts

```diff
@@ -239,21 +239,24 @@ export class DiagramView {
         this.source.trigger('updateWidgets', {widgets});
     }
 
-    showNavigationMenu(target: Element) {
+    showNavigationMenu(target: Element, onHide?: () => void) {
         const connectionsMenu = createElement(ConnectionsMenu, {
             view: this,
             target,
-            onClose: () => this.hideNavigationMenu(),
+            onClose: () => this.hideNavigationMenu(onHide),
             suggestProperties: this.options.suggestProperties,
         });
         this.connectionsMenuTarget = target;
         this.source.trigger('updateWidgets', {widgets: {connectionsMenu}});
     }
 
-    hideNavigationMenu() {
+    hideNavigationMenu(onHide?: () => void) {
         if (this.connectionsMenuTarget) {
             this.connectionsMenuTarget = undefined;
-            this.source.trigger('updateWidgets', {widgets: {connectionsMenu: undefined}});
+            this.source.trigger('updateWidgets', {widgets: { connectionsMenu: undefined}});
+            if (onHide) {
```

Comment:
I think instead of `onHide` callback we should add `toggleNavigationMenu` event (with `{ isOpen: boolean }` data) to `DiagramView` and subscribe to this event in the `DiagramView.configureHalo()` method.